tracing-journald/src/lib.rs

```diff
@@ -95,11 +97,14 @@ impl Subscriber {
         #[cfg(unix)]
         {
             let socket = UnixDatagram::unbound()?;
-            socket.connect("/run/systemd/journal/socket")?;
-            Ok(Self {
+            let sub = Self {
                 socket,
                 field_prefix: Some("F".into()),
-            })
+            };
+            // Check that we can talk to journald, by sending empty payload which journald discards.
+            // However if the socket didn't exist or if none listened we'd get an error here.
+            sub.send_payload(&[])?;
```

Comment:
@Ralith I'm not sure whether this is a good idea, but it seems to work as a simple check whether we can talk to journald, as you asked for in #1745 :slightly_smiling_face:

Reply:
I was concerned that journald might do something funny here, but [the shiny new journald protocol docs](https://systemd.io/JOURNAL_NATIVE_PROTOCOL/) explicitly say empty datagrams are ignored, so I think this is perfect.